aws/ecs/lib/resource/application.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ Copyright (c) 2016-2020, CKSource - Frederico Knabben. All rights reserved.
+ */
+
+import { Construct, Duration, Fn, Stack } from '@aws-cdk/core';
+import { ApplicationLoadBalancedEc2Service } from '@aws-cdk/aws-ecs-patterns';
+import {
+	Cluster,
+	ContainerDefinition,
+	Ec2TaskDefinition,
+	LaunchType,
+	LogDrivers,
+	PlacementStrategy,
+	Protocol,
+	RepositoryImage,
+	Secret,
+	TaskDefinition
+} from '@aws-cdk/aws-ecs';
+import { RetentionDays } from '@aws-cdk/aws-logs';
+import { InstanceClass, InstanceSize, InstanceType, Port } from '@aws-cdk/aws-ec2';
+import { ApplicationLoadBalancer, TargetGroupBase } from '@aws-cdk/aws-elasticloadbalancingv2';
+import { AutoScalingGroup } from '@aws-cdk/aws-autoscaling';
+
+import { Redis } from './redis';
+import { Database } from './database';
+import { Storage } from './storage';
+import { Network } from './network';
+import { StackConfig } from '../stack-config';
+import { CsEnvironmentConfig } from '../cs-environment-config';
+
+interface IApplicationProps {
+	stackConfig: StackConfig;
+	network: Network;
+	database: Database;
+	redis: Redis;
+	storage: Storage;
+	repositoryImage: RepositoryImage;
+}
+
+export class Application extends Construct {
+	private readonly taskDefinition: TaskDefinition;
+
+	private readonly containerDefinition: ContainerDefinition;
+
+	private readonly csEnvironmentConfig: CsEnvironmentConfig;
+
+	public readonly cluster: Cluster;
+
+	public readonly autoScalingGroup: AutoScalingGroup
+
+	private readonly loadBalancedEc2Service: ApplicationLoadBalancedEc2Service;
+
+	public constructor( scope: Construct, id: string, props: IApplicationProps ) {
+		super( scope, id );
+
+		this.csEnvironmentConfig = new CsEnvironmentConfig( this, props.stackConfig.env );
+
+		this.cluster = new Cluster( this, 'EcsCluster', { vpc: props.network.vpc } );
+
+		this.autoScalingGroup = this.cluster.addCapacity( 'DefaultAutoScalingGroupCapacity', {
+			instanceType: InstanceType.of( InstanceClass.C5, InstanceSize.LARGE ),
+			desiredCapacity: 2,
+			taskDrainTime: Duration.seconds( 0 )
+		} );
+
+		this.taskDefinition = new Ec2TaskDefinition( this, 'TaskDefinition' );
+
+		this.containerDefinition = this.taskDefinition.addContainer( 'CloudServicesContainer', {
+			image: props.repositoryImage,
+			environment: {
+				DATABASE_HOST: props.database.host,
+				DATABASE_DATABASE: props.database.name,
+				STORAGE_DRIVER: 's3',
+				STORAGE_ENDPOINT: props.storage.bucket.bucketDomainName,
+				STORAGE_BUCKET: props.storage.bucket.bucketName,
```

Comment:
IMO you can set `COLLABORATION_STORAGE_DRIVER` too.

https://ckeditor.com/docs/cs/latest/onpremises/cs-onpremises/installation/docker.html

```
STORAGE_DRIVER                           - optional (default: "filesystem")
STORAGE_LOCATION                         - optional (default: "/var/cs/easyimage", required for file system driver)
STORAGE_REGION                           - optional (required for AWS S3 driver)
STORAGE_ACCESS_KEY_ID                    - optional (required for AWS S3 driver)
STORAGE_SECRET_ACCESS_KEY                - optional (required for AWS S3 driver)
STORAGE_BUCKET                           - optional (required for AWS S3 driver)
STORAGE_ENDPOINT                         - optional (required for AWS S3 driver)
STORAGE_ACCOUNT_NAME                     - optional (required for Azure Blob Storage driver)
STORAGE_ACCOUNT_KEY                      - optional (required for Azure Blob Storage driver)
STORAGE_CONTAINER                        - optional (required for Azure Blob Storage driver)
COLLABORATION_STORAGE_DRIVER             - optional (default: "database")
COLLABORATION_STORAGE_LOCATION           - optional (default: "/var/cs/easyimage", required for file system driver)
COLLABORATION_STORAGE_ACCESS_KEY_ID      - optional (required for AWS S3 driver)
COLLABORATION_STORAGE_SECRET_ACCESS_KEY  - optional (required for AWS S3 driver)
COLLABORATION_STORAGE_REGION             - optional (required for AWS S3 driver)
COLLABORATION_STORAGE_ENDPOINT           - optional (required for AWS S3 driver)
COLLABORATION_STORAGE_BUCKET             - optional (required for AWS S3 driver)
COLLABORATION_STORAGE_ACCOUNT_NAME       - optional (required for Azure Blob Storage driver)
COLLABORATION_STORAGE_ACCOUNT_KEY        - optional (required for Azure Blob Storage driver)
COLLABORATION_STORAGE_CONTAINER          - optional (required for Azure Blob Storage driver)
```